addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -264,6 +264,15 @@ options.registry.gallery = options.Class.extend({
         });
         this._replaceContent($addImg.append($icon).append($text));
     },
+    /**
+     * @override
+     * TODO : remove me in master.
+     */
+    selectDataAttribute: function (previewMode, widget, params) {
+        const delay = params.activeValue.match(/[0-9]+/g);
+        this.$target[0].dataset.bsInterval = delay[0];
+        this._super(...arguments);
+    },
```

Comment:
Instead of patching the method behavior, you can patch the received template before it is used (see `_patchForComputeSnippetTemplates`)